### PR TITLE
able to append image searches to a div

### PR DESCRIPTION
**Pull request by imdanras**
**Deliverable Submission**

Please describe your comfort and completeness levels before submitting.

Comfort Level (1-5): **1**

Completeness Level (1-5): **1**

What did you think of this deliverable?: 😢 . Ouch. Incredibly difficult.

### CODE REVIEW

**Comment by scottheron** [comment]
Good job Dan, I like the style of the picture display.

File: src/js/app.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+$(document).ready(function() {
+
+
+
+
+$('#searchBtn').on('click', function(event) {
```
Comment (bhague1281) [inline src/js/app.js:6]: Note that for clicking on buttons, there's no default action. There is for forms though!

You may want to put an id on the form, then use a `'submit'` event instead. That way, when you press enter in the input field, it'll start the search.

File: src/js/app.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  $.get('https://www.reddit.com/search.json', {
+
+    q: searchInput,
+  }).done(function(data) {
+    intoArray(data);
+    // console.log(data);
+  }).fail(function(data) {
+    console.log('failed!');
+  });
```
Comment (bhague1281) [inline src/js/app.js:18]: Nice job adding some error handling. This is where you may also want to display a message to the user. But a console log works as a first step

File: src/js/app.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+function intoArray(data) {
+  var newArray = [];
+  var childrenArray = data.data.children;
+
+  for (var i = 0; i < childrenArray.length; i++) {
+    if (childrenArray[i].data.preview) {
+      //iterate over children[i]....url to push into a new array
+      var url = childrenArray[i].data.preview.images[0].source.url;
+      if (url !== '') {
```
Comment (bhague1281) [inline src/js/app.js:35]: Nice job creating a variable that represents the url. Very readable!

File: src/js/app.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
+
+});
+//function to grab the input from the text box and put it into the q field above
+function searchInput() {
+  return $('#input').val();
+}
+
+function intoArray(data) {
```
Comment (bhague1281) [inline src/js/app.js:27]: > There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.

You're guilty of the second one. What is `intoArray`? An alternative name for this function may be... `processImages`

Same goes with `doStuff`. An alternative name for that function may be... `startSlideshow`

File: src/js/app.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  $("#slideshow > div:gt(0)").hide();
+
+setInterval(function() {
+  $('#slideshow > div:first')
+    .fadeOut(1000)
+    .next()
+    .fadeIn(1000)
+    .end()
+    .appendTo('#slideshow');
+},  3000);
```
Comment (bhague1281) [inline src/js/app.js:60]: I've never seen this solution, but I like it! Iterating through each image and hiding/showing divs is a great solution.

**Comment by bhague1281** [comment]
Incredibly difficult, but you kept it clean and simple. Nice job!

**Comment by imdanras** [comment]
Thanks Brian, but full disclosure:

I had only figured out how to show the images in one giant div and that was
my original submission.  I got in about 8:30 this morning and googled
around to play with possible solutions, and came across that specific
setInterval solution. So, I pretty much copy/pasted it from the google
machine and played around/asked Scott until it looked better.

All that said, I still think it looks kinda cool.

**Pull request closed without merging**